hal/hal.spec

```diff
@@ -101,8 +101,8 @@
 @ stdcall -arch=i386,arm WRITE_PORT_UCHAR(ptr long)
 @ stdcall -arch=i386,arm WRITE_PORT_ULONG(ptr long)
 @ stdcall -arch=i386,arm WRITE_PORT_USHORT(ptr long)
-@ stdcall -arch=x86_64 x86BiosAllocateBuffer()
-@ stdcall -arch=x86_64 x86BiosCall()
-@ stdcall -arch=x86_64 x86BiosFreeBuffer()
-@ stdcall -arch=x86_64 x86BiosReadMemory()
-@ stdcall -arch=x86_64 x86BiosWriteMemory()
+@ stdcall -arch=i386,x86_64 x86BiosAllocateBuffer(ptr ptr ptr)
+@ stdcall -arch=i386,x86_64 x86BiosCall(long ptr)
+@ stdcall -arch=i386,x86_64 x86BiosFreeBuffer(long long)
+@ stdcall -arch=i386,x86_64 x86BiosReadMemory(long long ptr long)
+@ stdcall -arch=i386,x86_64 x86BiosWriteMemory(long long ptr long)
```

Comment:
We still need to keep them NT 5.2+ on AMD64, and I don't think we are able to export for different versions depending on architecture.

Reply:
While we're here, it would be nice to check other architectures. If it's all the same everywhere, `-arch:` directive could be just removed.

Reply:
> I don't think we are able to export for different versions depending on architecture.

Wouldn't something like this work?

```
@ stdcall -version=0x501-0x502 -arch=x86_64 x86BiosAllocateBuffer(ptr ptr ptr)
@ stdcall -version=0x600+ -arch=i386,x86_64 x86BiosAllocateBuffer(ptr ptr ptr)
```

Reply:
> While we're here, it would be nice to check other architectures. If it's all the same everywhere, -arch: directive could be just removed.

Seems to be NT 6.0+ on both x86 and AMD64. Win2k3 AMD64 has different exports:
![Screenshot_20230719_195317](https://github.com/reactos/reactos/assets/32551254/cd4a6dc4-9211-4ab9-99f9-fade8095c105)
(and also HalCallBios)
> Wouldn't something like this work?

Would probably clash on NT 6.0 AMD64 (I remember trying something like that a while ago).